frame/recovery/src/benchmarking.rs

```diff
@@ -0,0 +1,384 @@
+
+	friends.push(caller);
+
+	// Sort
+	friends.sort();
+
+	friends
+}
+
+fn insert_recovery_account<T: Config>(caller: &T::AccountId, account: &T::AccountId) {
+	T::Currency::make_free_balance_be(&account, BalanceOf::<T>::max_value());
+
+	let n = T::MaxFriends::get();
+
+	let friends = generate_friends::<T>(n);
+	let delay_period = block_number::<T>();
+
+	let bounded_friends: FriendsOf<T> = friends.try_into().unwrap();
+
+	// Get deposit for recovery
+	let total_deposit = get_total_deposit::<T>(&bounded_friends).unwrap();
+
+	let recovery_config = RecoveryConfig {
+		delay_period,
+		deposit: total_deposit,
+		friends: bounded_friends,
+		threshold: n as u16,
+	};
+
+	// Reserve deposit for recovery
+	T::Currency::reserve(&caller, total_deposit).unwrap();
+
+	<Recoverable<T>>::insert(&account, recovery_config);
+}
+
+benchmarks! {
+	as_recovered {
+		let caller: T::AccountId = whitelisted_caller();
+		let recovered_account: T::AccountId = account("recovered_account", 0, SEED);
+		let call: <T as Config>::Call = frame_system::Call::<T>::remark { remark: vec![] }.into();
+
+		Proxy::<T>::insert(&caller, &recovered_account);
+	}: _(
+		RawOrigin::Signed(caller),
+		recovered_account,
+		Box::new(call)
+	)
+
+	set_recovered {
+		let lost: T::AccountId = whitelisted_caller();
+		let rescuer: T::AccountId = whitelisted_caller();
+	}: _(
+		RawOrigin::Root,
+		lost.clone(),
+		rescuer.clone()
+	) verify {
+		assert_last_event::<T>(
+			Event::AccountRecovered {
+				lost_account: lost,
+				rescuer_account: rescuer,
+			}.into()
+		);
+	}
+
+	create_recovery {
+		let n in 1 .. T::MaxFriends::get();
+
+		let caller: T::AccountId = whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+
+		// Create friends
+		let friends = generate_friends::<T>(n);
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		friends,
+		n as u16,
+		42u32.into()
+	) verify {
+		assert_last_event::<T>(Event::RecoveryCreated { account: caller }.into());
+	}
+
+	initiate_recovery {
+		let caller: T::AccountId = whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+
+		let lost_account: T::AccountId = account("lost_account", 0, SEED);
+
+		insert_recovery_account::<T>(&caller, &lost_account);
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		lost_account.clone()
+	) verify {
+		assert_last_event::<T>(
+			Event::RecoveryInitiated {
+				lost_account: lost_account,
+				rescuer_account: caller,
+			}.into()
+		);
+	}
+
+	vouch_recovery {
+		let n in 1 .. T::MaxFriends::get();
+
+		let caller: T::AccountId = whitelisted_caller();
+		let lost_account: T::AccountId = account("lost_account", 0, SEED);
+		let rescuer_account: T::AccountId = account("rescuer_account", 0, SEED);
+
+		// Create friends
+		let friends = add_caller_and_generate_friends::<T>(caller.clone(), n);
+		let bounded_friends: FriendsOf<T> = friends.try_into().unwrap();
+
+		let delay_period = block_number::<T>();
```

Comment:
this doesn't really make sense to me. The delay should be some fixed, non-zero amount. In realistic benchmarks, this value coming from `block_number` will be 0. Is that intentional?

Reply:
Oh it wasn't, using `block_number` felt like a better option at the time, I will edit it to use a non-zero amount then.